### PR TITLE
remove claimconditions legacy

### PR DESCRIPTION
**Pull request by jnsdls**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to refactor claim condition components and improve error handling.

### Detailed summary
- Refactored claim condition components for better readability
- Improved error handling in `errorParser.tsx`
- Updated logic in `convert-erc20-amount.ts` for ERC20 token conversion
- Streamlined UI components in various files

> The following files were skipped due to too many changes: `apps/dashboard/src/contract-ui/tabs/claim-conditions/components/snapshot-upload.tsx`, `apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts`, `apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/index.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4590?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4590** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4590?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4588** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4588?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4587** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4587?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4586** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4586?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  return snapshot.map((entry) => {
+    if (typeof entry === "string") {
+      return {
+        address: entry,
+      } satisfies OverrideEntry;
+    }
+    return {
+      address: entry.address,
+      maxClaimable: toBigInt(entry.maxClaimable)?.toString(),
+      price: toBigInt(entry.price)?.toString(),
```
Comment (joaquim-verges) [inline apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts:164]: don't do toBigInt here, i already handle this in the SDK

File: apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+  return snapshot.map((entry) => {
+    if (typeof entry === "string") {
+      return {
+        address: entry,
+      } satisfies OverrideEntry;
+    }
+    return {
+      address: entry.address,
+      maxClaimable: toBigInt(entry.maxClaimable)?.toString(),
```
Comment (joaquim-verges) [inline apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts:163]: same here

File: apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  if (timestamp instanceof Date) {
+    return timestamp;
+  }
+  return new Date(timestamp);
+}
+function toBigInt(value: string | number | undefined) {
+  if (value === undefined) {
+    return undefined;
+  }
+  if (value === "unlimited") {
```
Comment (joaquim-verges) [inline apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts:182]: i already handle this internally, you should not have to convert anything yourself

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**